Add custom styles to iOS SearchBar cancel button

tux-tn opened the pull request:
This PR adds two attributes to ```cancelbuttonprops``` in SearchBar IOS: 

- ```buttonStyle``` :  Cancel button custom styles
- ```buttonTextStyle``` : Cancel button's text custom styles

 The main goal is to allow customization of cancel button since react-native's ```Button``` doesn't allow that

iRoachie commented:
Hey thanks for sending the PR. i think we were going to tackle this by using the Button component from RNE. @xavier-villelegier can you confirm?

tux-tn commented:
@iRoachie do you want me to update my PR and use RNE Button or should i close it? BTW, i am using the exact same styling as RN ```Button``` component

iRoachie commented:
@xavier-villelegier

xavier-villelegier commented:
I think it's okay to use a simple TouchableOpacity for now, we don't need all the Button's props.
But maybe we should use the `primary` color of the theme ?

tux-tn commented:
@xavier-villelegier Don't you think we should keep default iOS color and it's material design equivalent since it's an iOS searchbar?

xavier-villelegier commented:
@tux-tn Actually, as you said it's an iOS SearchBar, so I was just wondering if the Button should look different on iOS and Android ? I mean, why should the SearchBar be all the same accross platforms except the button ? 🤔 Maybe we should just keep the iOS style on both platforms (cc @iRoachie)

iRoachie commented:
Sounds good to me. A change needs to be made anyway cause right now on android the cancel button is the big blue material raised button

xavier-villelegier commented:
Yea that's why @tux-tn work is necessary to make it consistent between platforms, actually you just need to remove the `Platform.select` from the styles and keep the iOS one

tux-tn commented:
I agree with that, i will make the necessary changes

xavier-villelegier commented on src/searchbar/SearchBar-ios.js:
```diff
@@ -155,11 +164,33 @@ class SearchBar extends Component {
+            <View
+              style={[
+                buttonColor &&
+                  Platform.OS === 'android' && { backgroundColor: buttonColor },
```
Actually as we said you shouldn't have any differences between iOS and Android here (no background on iOS, neither on Android)

xavier-villelegier commented on src/searchbar/SearchBar-ios.js:
```diff
@@ -155,11 +164,33 @@ class SearchBar extends Component {
+                style={[
+                  styles.buttonTextStyle,
+                  buttonColor &&
+                    Platform.OS === 'ios' && { color: buttonColor },
```
Same goes for this one, both iOS and Android should have this color

tux-tn commented:
The reason i kept this conditional styling is to follow RN Button documentation. We need to either keep that or change searchbar-ios documentation to stop referencing RN Button props as props for cancel button. What do you think ? 



Le 18 nov. 2018 à 1:56 PM, à 1:56 PM, "Xavier Villelégier" <notifications@github.com> a écrit:
>xavier-villelegier requested changes on this pull request.
>
>Almost there ! Just few style changes to make it consistent on both
>platforms ✨
>
>>              onPress={this.cancel}
>-            {...cancelButtonProps}
>-          />
>+            disabled={buttonDisabled}
>+            {...otherCancelButtonProps}
>+          >
>+            <View
>+              style={[
>+                buttonColor &&
>+                  Platform.OS === 'android' && { backgroundColor:
>buttonColor },
>
>Actually as we said you shouldn't have any differences between iOS and
>Android here (no background on iOS, neither on Android)
>
>> -          />
>+            disabled={buttonDisabled}
>+            {...otherCancelButtonProps}
>+          >
>+            <View
>+              style={[
>+                buttonColor &&
>+                  Platform.OS === 'android' && { backgroundColor:
>buttonColor },
>+                buttonStyle,
>+              ]}
>+            >
>+              <Text
>+                style={[
>+                  styles.buttonTextStyle,
>+                  buttonColor &&
>+                    Platform.OS === 'ios' && { color: buttonColor },
>
>Same goes for this one, both iOS and Android should have this color
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/react-native-training/react-native-elements/pull/1449#pullrequestreview-176088299

xavier-villelegier commented:
Yea I see what you mean, let's update the doc then. The iOS SearchBar with an android `Button` looks weird in my opinion, it's better to keep all the iOS styling for the "ios" SearchBar

tux-tn commented:
I removed Platform specific styling. I also added props for disabled cancel Button styling (The main purpose of my PR was to add the ability to customize the cancel Button).

tux-tn commented:
Thank you @xavier-villelegier, should i merge next updates and resolve conflicts?

xavier-villelegier commented:
Let's merge that and wait for some feedbacks to polish it.

Thanks @tux-tn, congrats for your first-time contribution to RNE 👏 🥇

iRoachie commented:
Just realize we merged this with adding typescript definitions...

iRoachie commented:
Tracked over here https://github.com/react-native-training/react-native-elements/issues/1606
